### ISSUE TITLE
trailing slash in MODULEPATH dirs removes first character from module names

### ISSUE THREAD

**Issue by ghuls** [cea-hpc/modules#152]
trailing slash in MODULEPATH dirs removes first character from each module name.

Tested with **fish**
```fish
$ set MODULEPATH /apps/leuven/etc/modules/

# Module names are missing the first character
$ module avail
--------------------------------------------------------------------------- /apps/leuven/etc/modules/ ----------------------------------------------------------------------------
20Xm/2014a  20Xm/2015a  40c/2014a  40c/2015a  2070/2014a  ctstest/2014a  erebro-old  erebro/2014a  hi/2014a  hinking/2014a  hinking2/2014a
```

```fish
$ set MODULEPATH /apps/leuven/etc/modules

# Module names are correct
$ module avail
---------------------------------------------------------------------------- /apps/leuven/etc/modules ----------------------------------------------------------------------------
cerebro-old  cerebro/2014a  ictstest/2014a  K20Xm/2014a  K20Xm/2015a  K40c/2014a  K40c/2015a  M2070/2014a  phi/2014a  thinking/2014a  thinking2/2014a
```

**Comment by xdelaruelle** [comment]
Oh! quite bad. Need to better check MODULEPATH variable content, as it can be set by hand.

Until being fixed,``module use`` should be used rather setting directly MODULEPATH.

Thanks for the report.

**Comment by ghuls** [comment]
Hi Xavier.

I discovered the bug because I wanted to use fish on the cluster I am using
but the installed module system does not support fish yet, so I tried to
use the last version of modules. The old version of the modules system had
paths with a trailing slash, so when switching to the newer version, the
paths with trailing slashes caused problems.

Greetings,
Gert

On Jan 26, 2018 8:02 PM, "Xavier Delaruelle" <notifications@github.com>
wrote:

Oh! quite bad. Need to better check MODULEPATH variable content, as it can
be set by hand.

Until being fixed,module use should be used rather setting directly
MODULEPATH.

Thanks for the report.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/cea-hpc/modules/issues/152#issuecomment-360874421>, or mute
the thread
